### PR TITLE
#187300235 Verify user email after registration

### PR DESCRIPTION
**Pull request by sabatohura**
#### What does this PR do?

- Verifying user account email after registration

#### Description of Task to be completed?

- Setting Up mailer
- Setting mailer templates
- Send token and URI for verification after registration
- Verify User email

#### How should this be manually tested?
- In normal registration api URL, [baseUrl/api/users/signup]
- Create account
- Check email for verification
- Click verification link in email

### CODE REVIEW

**Review by ErnestTchami — APPROVED** [review]

**Review by amiparadis250 — APPROVED** [review]

**Review by istafrica-dennw — CHANGES_REQUESTED** [review]
1. add a label
2. let us add a coverage check in our checks

File: src/__test__/initial.spec.ts [hunk 1]
```diff
@@ -70,8 +65,19 @@ describe("test a user signup endpoint", () => {
         expect(res).to.have.status(400);
       });
   });
-});
 
+  // it("user fail to verify user with invalid token", () => {
```
Comment (Eli250) [inline src/__test__/initial.spec.ts]: Why did you comment out this test case?

File: src/controllers/verify.controller.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { config } from "dotenv";
+import { Request, Response } from "express";
+import jwt from "jsonwebtoken";
+import User from "../models/user";
+
+config();
+
+export const verifyAccount = async (req: Request, res: Response) => {
```
Comment (Eli250) [inline src/controllers/verify.controller.ts]: This operation should be under user controller not a separate controller. Move it under user controllers or make it as a middleware.

File: src/models/user.ts [hunk 3]
```diff
@@ -66,13 +67,9 @@ User.init(
       allowNull: true,
       defaultValue: ""
     },
-    createdAt: {
```
Comment (Eli250) [inline src/models/user.ts:69]: Why did you remove the time stamp for this model?

File: src/utils/encrypt.ts [hunk 4]
```diff
@@ -14,3 +14,14 @@ export const passwordCompare = async (
   const verifyPassword = await bcrypt.compare(inputPassword, userPassword);
   return verifyPassword;
 };
+
```
Comment (Eli250) [inline src/utils/encrypt.ts]: Why did you comment these utility functions?

**Review by Eli250 — CHANGES_REQUESTED** [review]
- Resolve all feedback
- Increase your test coverage.

**Review by istafrica-dennw — APPROVED** [review]

File: src/__test__/initial.spec.ts (hunk 1 above)
Comment (sabatohura) [reply src/__test__/initial.spec.ts]: I brought it back, and now it is working well. i was checking why it was not working on circle ci but now i have fixed it.

File: src/models/user.ts (hunk 3 above)
Comment (sabatohura) [reply src/models/user.ts:69]: there I removed the timestamp to use the timestamp from sequelize, those codes were doing the same thing as a timestamp: true which was in that sequelize options object below

File: src/controllers/verify.controller.ts (hunk 2 above)
Comment (sabatohura) [reply src/controllers/verify.controller.ts]: because everyone was working under the same file I was minimizing conflicts. but I can take it back in user controllers.

File: src/utils/encrypt.ts (hunk 4 above)
Comment (sabatohura) [reply src/utils/encrypt.ts]: I was using those functions to encrypt and decrypt user data before while verifying the user, but I switched to jwt I was supposed to remove them, but now I have removed them
